src/test/java/com/google/sps/AbuseDetectionTest.java

```diff
@@ -2,19 +2,231 @@
+    value = instance.addRequest(inputTime);
+
+    inputTime = LocalTime.parse("23:59:55.344");
+    value = instance.addRequest(inputTime);
+    
+    inputTime = LocalTime.parse("23:59:55.421");
+    value = instance.addRequest(inputTime);
+    
+    inputTime = LocalTime.parse("23:59:55.560");
+    value = instance.addRequest(inputTime);
+
+    inputTime = LocalTime.parse("23:59:55.590");
+    value = instance.addRequest(inputTime);
+    
+    inputTime = LocalTime.parse("23:59:55.612");
+    value = instance.addRequest(inputTime);
+
+    inputTime = LocalTime.parse("23:59:55.642");
+    value = instance.addRequest(inputTime);
+   
+    inputTime = LocalTime.parse("23:59:55.723");
+    value = instance.addRequest(inputTime);
+
+    inputTime = LocalTime.parse("23:59:55.800");
+    value = instance.addRequest(inputTime);
+
+    inputTime = LocalTime.parse("23:59:55.900");
+    value = instance.addRequest(inputTime);
+   
+    return value;
+  }
+
+  @Test
+  public void addingOneTimeRequest() {
+    AbuseDetection instance = new AbuseDetection(1, 10);
+    LocalTime inputTime = LocalTime.parse("18:19:20.010");
+    boolean value = instance.addRequest(inputTime);
+
+    Assert.assertEquals(true, value);
+  }
+
+  @Test
+  public void addingMultipleRequestsGetFalse() {
+    AbuseDetection instance = new AbuseDetection(1, 10);
+    boolean returnValue = addMultipleRequests(instance);
+    LocalTime inputTime = LocalTime.parse("18:19:20.901");
+    boolean value = instance.addRequest(inputTime);
+
+    Assert.assertEquals(false, value);
+  }
+
+  @Test
+  public void addingMultipleRequestsGetTrue() {
+    AbuseDetection instance = new AbuseDetection(1, 10);
+    boolean returnValue = addMultipleRequests(instance);
+    LocalTime inputTime = LocalTime.parse("18:19:21.012");
+    boolean value = instance.addRequest(inputTime);
+
+    Assert.assertEquals(true, value);
+  }
+
+  @Test
+  public void requestOverSecondReturnsFalse() {
+    AbuseDetection instance = new AbuseDetection(1, 10);
+    boolean returnValue = addMultipleRequests(instance);
+    LocalTime inputTime1 = LocalTime.parse("18:19:21.012");
+    boolean curVal = instance.addRequest(inputTime1);
+    LocalTime otherTime = LocalTime.parse("18:19:21.109");
+    boolean value = instance.addRequest(otherTime);
+
+    Assert.assertEquals(false, value);
+  }
+  
+
+  @Test
+  public void requestOverSecondReturnsTrue() {
+    AbuseDetection instance = new AbuseDetection(1, 10);
+    boolean returnValue = addMultipleRequests(instance);
+    LocalTime inputTime1 = LocalTime.parse("18:19:21.012");
+    boolean curVal = instance.addRequest(inputTime1);
+    LocalTime otherTime = LocalTime.parse("18:19:21.109");
+    boolean valueReturned = instance.addRequest(otherTime);
+    LocalTime timeInputted = LocalTime.parse("18:19:21.416");
+    boolean value = instance.addRequest(timeInputted);
+
+    Assert.assertEquals(true, value);
+  }
+
+  @Test
+  public void addingRequestsNearMidnightTest() {
```

Comment:
Could you a  test that crosses the midnight boundary?

Reply:
What specifically do you mean? On Line 205 I have a test that crosses the midnight boundary.

Are you asking to see another case where 10 requests are sent and few of them cross into the midnight boundary?

Reply:
okay - I didn't realize this test crossed the midnight boundary.

So it looks you send 20 requests before midnight and then you send a single request 20 minutes later. The last request should not have exceeded the limit. Why is it being dropped in this test?

Reply:
Umm. Oh I mistyped that other request. I'll commit the right time for it. Also, I trimmed it down to have less requests added.

Reply:
So after changing the amount of requests coming in to 10, the 11th request is being dropped because 10 requests have already been stored in.

Reply:
10 were made**